docs/tutorial/part_2.rst

```diff
@@ -0,0 +1,459 @@
+    ]
+
+The next step is to point the root routing configuration at the **chat.routing**
+module. In **mysite/routing.py**, import ``AuthMiddlewareStack``, ``URLRouter``,
+and ``chat.routing``; and insert a ``'websocket'`` key in the
+``ProtocolTypeRouter`` list in the following format::
+
+    # mysite/routing.py
+    from channels.auth import AuthMiddlewareStack
+    from channels.routing import ProtocolTypeRouter, URLRouter
+    import chat.routing
+    
+    application = ProtocolTypeRouter({
+        # (http->django views is added by default)
+        'websocket': AuthMiddlewareStack(
+            URLRouter(
+                chat.routing.websocket_urlpatterns
+            )
+        ),
+    })
+
+This root routing configuration specifies that when a connection is made to the
+Channels development server, the ``ProtocolTypeRouter`` will first inspect the type
+of connection. If it is a WebSocket connection (**ws://** or **wss://**), the connection
+will be given to the ``AuthMiddlewareStack``.
+
+The ``AuthMiddlewareStack`` will populate the connection’s **scope** with a reference to
+the currently authenticated user, similar to how Django’s
+``AuthenticationMiddleware`` populates the **request** object of a view function with
+the currently authenticated user. (Scopes will be discussed later in this
+tutorial.) Then the connection will be given to the ``URLRouter``.
+
+The ``URLRouter`` will examine the HTTP path of the connection to route it to a
+particular consumer, based on the provided ``url`` patterns.
+
+Let’s verify that the consumer for the ``/ws/chat/ROOM_NAME/`` path works. Start the
+Channels development server::
+
+    $ python3 manage.py runserver
+
+Go to the room page at http://127.0.0.1:8000/chat/lobby/ which now displays an
+empty chat log.
+
+Type the message "hello" and press enter. You should now see "hello" echoed in
+the chat log.
+
+However if you open a second browser tab to the same room page at
+http://127.0.0.1:8000/chat/lobby/ and type in a message, the message will not
+appear in the first tab. For that to work, we need to have multiple instances of
+the same ``ChatConsumer`` be able to talk to each other. Channels provides a
+**channel layer** abstraction that enables this kind of communication between
+consumers.
+
+Go to the terminal where you ran the **runserver** command and press Control-C to
+stop the server.
+
+Enable a channel layer
+----------------------
+
+A channel layer is a kind of communication system. It allows multiple consumer
+instances to talk with each other, and with other parts of Django.
+
+A channel layer provides the following abstractions:
+
+* A **channel** is a mailbox where messages can be sent to. Each channel has a name.
+  Anyone who has the name of a channel can send a message to the channel.
+
+* A **group** is a group of related channels. A group has a name. Anyone who has the
+  name of a group can (1) add/remove a channel to the group by name and (2) send
```

Comment:
Can we format this either as a nested list or just drop the numbers and use a comma to separate them?